Support creating PRs in other repos

### Subject of the issue

It would be ideal to support creating PRs (using a custom personal access token) to remote repositories, using the `git remote` information.

### Steps to reproduce

Given a workflow like:

```yml
on: [push]

jobs:
  create-pr-on-remote-repo:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          token: ${{ secrets.PERSONAL_TOKEN }}
          repository: remote_owner/remote_repo
      - run: 'echo "v1.0" > .version'
      - uses: peter-evans/create-pull-request@v2.2.0
        with:
          token: ${{ secrets.PERSONAL_TOKEN }}
          branch: auto-update-workflows-docs
```

I would expect:

✅ Creates PR in remote repository

but

❌ Tries to create PR in workflow's repo:

```
Pull request branch to create or update set to 'create-pull-request/patch'
Configured git committer as 'GitHub <noreply@github.com>'
Configured git author as 'github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>'
Uncommitted changes found. Adding a commit.
Pull request branch 'create-pull-request/patch' does not exist yet
Created branch 'create-pull-request/patch'
Pushing pull request branch to 'origin/create-pull-request/patch'

Traceback (most recent call last):
  File "/home/runner/work/_actions/peter-evans/create-pull-request/v2.2.0/dist/src/create_pull_request.py", line 174, in <module>
    repo.git.push("--force", repo_url, f"HEAD:refs/heads/{branch}")
  File "/opt/hostedtoolcache/Python/3.8.1/x64/lib/python3.8/site-packages/git/cmd.py", line 551, in <lambda>
    return lambda *args, **kwargs: self._call_process(name, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.1/x64/lib/python3.8/site-packages/git/cmd.py", line 1020, in _call_process
    return self.execute(call, **exec_kwargs)
  File "/opt/hostedtoolcache/Python/3.8.1/x64/lib/python3.8/site-packages/git/cmd.py", line 831, in execute
    raise GitCommandError(command, status, stderr_value, stdout_value)

git.exc.GitCommandError: Cmd('git') failed due to: exit code(1)
  cmdline: git push --force ***github.com/current_owner/current_repo HEAD:refs/heads/create-pull-request/patch
  stderr: 'To https://github.com/current_owner/current_repo
 ! [remote rejected] HEAD -> create-pull-request/patch (shallow update not allowed)

error: failed to push some refs to '***github.com/current_owner/current_repo''

##[error]The process '/opt/hostedtoolcache/Python/3.8.1/x64/bin/python' failed with exit code 1

##[error]Node run failed with exit code 1
```

--- a/src/common.py
+++ b/src/common.py
@@ -8,6 +8,19 @@ def get_random_string(length=7, chars=string.ascii_lowercase + string.digits):
     return "".join(random.choice(chars) for _ in range(length))
 
 
+def parse_github_repository(url):
+    # Parse the github repository from a URL
+    # e.g. peter-evans/create-pull-request
+    pattern = re.compile(r"^https://github.com/(.+/.+)$")
+
+    # Check we have a match
+    match = pattern.match(url)
+    if match is None:
+        raise ValueError(f"The format of '{url}' is not a valid GitHub repository URL")
+
+    return match.group(1)
+
+
 def parse_display_name_email(display_name_email):
     # Parse the name and email address from a string in the following format
     # Display Name <email@address.com>

--- a/src/create_pull_request.py
+++ b/src/create_pull_request.py
@@ -31,6 +31,13 @@ def get_git_config_value(repo, name):
         return None
 
 
+def get_github_repository():
+    remote_origin_url = get_git_config_value(repo, "remote.origin.url")
+    if remote_origin_url is None:
+        raise ValueError("Failed to fetch 'remote.origin.url' from git config")
+    return cmn.parse_github_repository(remote_origin_url)
+
+
 def git_user_config_is_set(repo):
     name = get_git_config_value(repo, "user.name")
     email = get_git_config_value(repo, "user.email")
@@ -94,7 +101,6 @@ def set_committer_author(repo, committer, author):
 
 # Get required environment variables
 github_token = os.environ["GITHUB_TOKEN"]
-github_repository = os.environ["GITHUB_REPOSITORY"]
 # Get environment variables with defaults
 path = os.getenv("CPR_PATH", os.getcwd())
 branch = os.getenv("CPR_BRANCH", DEFAULT_BRANCH)
@@ -106,6 +112,11 @@ base = os.environ.get("CPR_BASE")
 
 # Set the repo path
 repo = Repo(path)
+
+# Determine the GitHub repository from git config
+# This will be the target repository for the pull request
+github_repository = get_github_repository()
+print(f"Target repository set to {github_repository}")
 
 # Determine if the checked out ref is a valid base for a pull request
 # The action needs the checked out HEAD ref to be a branch

--- a/src/test_common.py
+++ b/src/test_common.py
@@ -9,6 +9,23 @@ def test_get_random_string():
     assert len(cmn.get_random_string(length=20)) == 20
 
 
+def test_parse_github_repository_success():
+    repository = cmn.parse_github_repository(
+        "https://github.com/peter-evans/create-pull-request"
+    )
+    assert repository == "peter-evans/create-pull-request"
+
+
+def test_parse_github_repository_failure():
+    url = "https://github.com/peter-evans"
+    with pytest.raises(ValueError) as e_info:
+        cmn.parse_github_repository(url)
+    assert (
+        e_info.value.args[0]
+        == f"The format of '{url}' is not a valid GitHub repository URL"
+    )
+
+
 def test_parse_display_name_email_success():
     name, email = cmn.parse_display_name_email("abc def <abc@def.com>")
     assert name == "abc def"

